src/app/_components/chats/ImageUploadDialog.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactNode, useState } from "react";
+import { Dialog, DialogContent, DialogTrigger } from "@components/dialog";
+import { PenSquareIcon } from "lucide-react";
+import ImageUpload from "@components/../image-uploader";
+
+interface ImageUploadDialogProps {
+  onUploadSuccess: (url: string) => void;
+  Button : ReactNode
+}
+
+const ImageUploadDialog: React.FC<ImageUploadDialogProps> = ({
+  onUploadSuccess,
+  Button
+}: ImageUploadDialogProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <Dialog open={isOpen} onOpenChange={setIsOpen}>
+      <DialogTrigger asChild>
+      {Button}
+      </DialogTrigger>
+      <DialogContent>
+        <ImageUpload
+          onUploadSuccess={(url) => {
+            onUploadSuccess(url); // Call the parent's function
+            setIsOpen(false); // Close the dialog
+          }}
```

Comment:
move it to upper function called handleUploadSuccess and then it'll be in the html
<ImageUpload onUploadSuccess={handleUploadSuccess}/>

Reply:
Done